### ISSUE TITLE
Add note about `toStringTag`

### ISSUE THREAD

**Issue by bakkot** [inspect-js/is-arguments#20]
This gets fooled by

```js
({ [Symbol.toStringTag]: 'Arguments' })
```

a fact which should probably be noted in the readme.

**Comment by ljharb** [comment]
Is there any way you're aware of to be more robust with that check?

**Comment by bakkot** [comment]
I'm not aware of any way to be perfectly robust in all circumstances. In practice, `arguments` objects will not tend to have their prototype or `toStringTag` property modified, so if you're willing to accept some false negatives in exchange for fewer false positives you can use

```js
toStr.call(value) === '[object Arguments]' && (Object.getPrototypeOf(value) === Object.prototype || Object.getPrototypeOf(value) === null) && !Object.prototype.hasOwnProperty.call(value, Symbol.toStringTag)
```

**Comment by ljharb** [comment]
hm, when would `Object.getPrototypeOf(value) === null` be true?

**Comment by ljharb** [comment]
Of note, in latest Safari, arguments objects do not appear to have a `Symbol.toStringTag` property at all - which means the toString is the only way to be sure :-/

**Comment by bakkot** [comment]
> hm, when would `Object.getPrototypeOf(value) === null` be true?

```js
(function(){ Object.setPrototypeOf(arguments, null); })()
```
> arguments objects do not appear to have a `Symbol.toStringTag` property at all

No arguments objects do, unless someone's explicitly added one. That's actually nice: it means that 
- if neither `value` nor anything in its prototype chain has a `Symbol.toStringTag`, then
- `Object.prototype.toString.call(value)` returns `[object Arguments]` iff `value` is an arguments object. 

If the first property does not hold (and cannot be made to hold), you can never be sure.

**Comment by ljharb** [comment]
aha, ok so `return !(Symbol.toStringTag in value) && Object.prototype.toString.call(value) === '[object Arguments]'` - and then to be sure it's not an `Object.create(arguments)` - checking to ensure that the [[Prototype]] is `Object.prototype` wouldn't work cross-realm, and checking that the proto is null wouldn't work for stock arguments objects, so i'm not sure that's a particularly useful check.

**Comment by bakkot** [comment]
> then to be sure it's not an `Object.create(arguments) `

No need for that; such objects do not appear as `[object Arguments]`.

There's no real need for the prototype checks; forget I mentioned them.
